### PR TITLE
Added ActivityRegistry signals for EDProducer transforms

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

If an EDProducer registers transform callbacks, the ActivityRegistry will not emit signals for
- prefetching data needed for the transform
- calling of any asynchronous pre-transform task
- calling of the transform task The Tracer service was updated to use the new signals.

#### PR validation:

Code compiles. Framework unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/src/WorkerT.cc [hunk 1]
```diff
@@ -252,8 +252,8 @@ namespace edm {
       ServiceRegistry::Operate guard(weakToken.lock());
 
       ModuleCallingContext mcc(
-          &module_->moduleDescription(), ModuleCallingContext::State::kPrefetching, iParent, nullptr);
-      module_->doTransformAsync(iTask, iTransformIndex, iEvent, activityRegistry(), &mcc, weakToken);
+          &module_->moduleDescription(), iTransformIndex, ModuleCallingContext::State::kRunning, iParent, nullptr);
```
Comment (makortel) [inline FWCore/Framework/src/WorkerT.cc:255]: Do I understand correctly that the state should have been set to `kRunning` already before?

Comment (Dr15Jones) [reply FWCore/Framework/src/WorkerT.cc:255]: yes

File: FWCore/Services/plugins/Tracer.cc [hunk 2]
```diff
@@ -1228,6 +1240,87 @@ void Tracer::postEventReadFromSource(StreamContext const& sc, ModuleCallingConte
...
+
+void Tracer::postModuleTransformPrefetching(StreamContext const& sc, ModuleCallingContext const& mcc) {
+  LogAbsolute out("Tracer");
+  out << TimeStamper(printTimestamps_);
+  unsigned int nIndents = mcc.depth() + 4;
+  for (unsigned int i = 0; i < nIndents; ++i) {
+    out << indention_;
+  }
+  out << " finished: prefetching before transform in event for module: stream = " << sc.streamID() << " label = '"
+      << mcc.moduleDescription()->moduleLabel() << "' id = " << mcc.moduleDescription()->id();
```
Comment (makortel) [inline FWCore/Services/plugins/Tracer.cc:1266]: Would it make sense to add the `mcc.callID()` to the transformer printouts?

Comment (Dr15Jones) [reply FWCore/Services/plugins/Tracer.cc:1266]: Yes, definitely.

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Services/plugins/Tracer.cc (hunk 2 above)
Comment (Dr15Jones) [reply FWCore/Services/plugins/Tracer.cc]: info added to printout.

**Comment by makortel** [comment]
+core

**Pull request merged**